packages/core/src/app/bootstrap.ts

```diff
@@ -69,57 +115,38 @@ export async function editorBootstrap(
   settingsFile: MetaFile<any>,
 ): Promise<Project> {
   const logger = config.logger ?? new Logger();
-  const promises: Promise<Plugin | null>[] = [Promise.resolve(DefaultPlugin())];
+  const pluginResolutions = new Map<string, Plugin>();
 
-  if (config.plugins) {
-    for (const plugin of config.plugins) {
-      promises.push(parsePlugin(plugin, logger));
-    }
+  async function resolvePlugin(plugin: Plugin | string) {
```

Comment:
These functions seem almost the same as the ones above, but have a couple differences. Could we consolidate them with exported functions, or just a top level function that is not stateful (i.e., takes in the resolution list and plugin, and returns the plugin or undefined)?